web-common/src/features/dashboards/filters/measure-filters/MeasureFilterMenu.svelte

```diff
@@ -0,0 +1,170 @@
+<script lang="ts">
+  import InputV2 from "@rilldata/web-common/components/forms/InputV2.svelte";
+  import Select from "@rilldata/web-common/components/forms/Select.svelte";
+  import { Menu } from "@rilldata/web-common/components/menu";
+  import { getDimensionDisplayName } from "@rilldata/web-common/features/dashboards/filters/getDisplayName";
+  import { MeasureFilterOptions } from "@rilldata/web-common/features/dashboards/filters/measure-filters/measure-filter-options";
+  import { getStateManagers } from "@rilldata/web-common/features/dashboards/state-managers/state-managers";
+  import {
+    createBetweenExpression,
+    createBinaryExpression,
+  } from "@rilldata/web-common/features/dashboards/stores/filter-utils";
+  import {
+    type V1Expression,
+    V1Operation,
+  } from "@rilldata/web-common/runtime-client";
+  import { createEventDispatcher } from "svelte";
+  import { createForm } from "svelte-forms-lib";
+  import * as yup from "yup";
+
+  export let dimensionName: string;
+  export let name: string;
+  export let expr: V1Expression | undefined;
+
+  const dispatch = createEventDispatcher();
+
+  const {
+    selectors: {
+      dimensions: { allDimensions },
+    },
+  } = getStateManagers();
+
+  $: dimensionOptions =
+    $allDimensions?.map((d) => ({
+      value: d.name as string,
+      label: getDimensionDisplayName(d),
+    })) ?? [];
+
+  let operation: string = MeasureFilterOptions[0].value;
+  let value1;
+  let value2;
+  if (!expr?.cond?.op) {
+    operation = MeasureFilterOptions[0].value;
+    value1 = undefined;
+    value2 = undefined;
+  } else {
+    if (expr?.cond?.op === V1Operation.OPERATION_AND) {
+      operation = "b";
+      value1 = expr?.cond?.exprs?.[0].cond?.exprs?.[1]?.val as string;
+      value2 = expr?.cond?.exprs?.[1].cond?.exprs?.[1]?.val as string;
+    } else if (expr?.cond?.op === V1Operation.OPERATION_OR) {
+      operation = "nb";
+      value1 = expr?.cond?.exprs?.[0].cond?.exprs?.[1]?.val as string;
+      value2 = expr?.cond?.exprs?.[1].cond?.exprs?.[1]?.val as string;
+    } else {
+      operation = expr?.cond?.op;
+      value1 = expr?.cond?.exprs?.[1]?.val as string;
+    }
+  }
+
+  const formState = createForm({
+    initialValues: {
+      dimension: dimensionName,
+      operation,
+      value1,
+      value2,
+    },
+    validationSchema: yup.object({
+      dimension: yup.string().required("Required"),
+      operation: yup.string().required("Required"),
+      value1: yup.number().required(),
```

Comment:
NIT: Form error validation looks kinda ugly as it takes the property name

![image](https://github.com/rilldata/rill/assets/1181922/ff319f13-2e29-4c10-9604-b0c46241a696)